scripts/setup-database.sh

```diff
@@ -3,26 +3,76 @@
 source /scripts/env-data.sh
 
 POSTGRES_PASS=$(cat /tmp/PGPASSWORD.txt)
+INITDB_WALDIR_FLAG=""
+
+# Check POSTGRES_INITDB_WALDIR value
+if [[ -n "${POSTGRES_INITDB_WALDIR}" ]]; then
+    # If POSTGRES_INITDB_WALDIR is defined, make sure that it is not inside 
+    # the ${DATADIR} directory, to avoid deletions
+    case "${POSTGRES_INITDB_WALDIR}" in
+        ${DATADIR}/*)
+            # In this case, we have to fail early
+            echo "POSTGRES_INITDB_WALDIR should not be set to be inside DATADIR or PGDATA" 
+cat << EOF 1>&2
+Error!
+POSTGRES_INITDB_WALDIR should not be set to be inside DATADIR or PGDATA.
+POSTGRES_INITDB_WALDIR: ${POSTGRES_INITDB_WALDIR}
+DATADIR or PGDATA: ${DATADIR}
+EOF
+            exit 1
+            ;;
+        *)
+            # For other case, make sure the directory is created with proper permissions
+            create_dir "${POSTGRES_INITDB_WALDIR}"
+            chown -R postgres:postgres ${POSTGRES_INITDB_WALDIR}
+            ;;
+    esac
+    # Set the --waldir flag for postgres initialization
+    INITDB_WALDIR_FLAG="--waldir ${POSTGRES_INITDB_WALDIR}"
+fi
 
 # test if DATADIR has content
-# Do initialization if DATADIR is empty, or RECREATE_DATADIR is true
-if [[ -z "$(ls -A ${DATADIR} 2> /dev/null)" || "${RECREATE_DATADIR}" == 'TRUE' ]]; then
+# Do initialization if DATADIR directory is empty, or RECREATE_DATADIR is true
+if [[ -z "$(ls -A ${DATADIR} 2> /dev/null)" || "${RECREATE_DATADIR}" =~ [Tt][Rr][Uu][Ee] ]]; then
     # Only attempt reinitializations if ${RECREATE_DATADIR} is true
     # No Replicate From settings. Assume that this is a master database.
     # Initialise db
     echo "Initializing Postgres Database at ${DATADIR}"
-    create_dir ${POSTGRES_INITDB_WALDIR}
-    create_dir ${DATADIR}
+    create_dir "${DATADIR}"
     rm -rf ${DATADIR}/*
-    chown -R postgres:postgres ${DATADIR} ${POSTGRES_INITDB_WALDIR}
+    chown -R postgres:postgres "${DATADIR}"
     echo "Initializing with command:"
-    command="$INITDB -U postgres --pwfile=<(echo "$POSTGRES_PASS") -E ${DEFAULT_ENCODING} --lc-collate=${DEFAULT_COLLATION} --lc-ctype=${DEFAULT_CTYPE} --wal-segsize=${WAL_SEGSIZE} --auth=${PASSWORD_AUTHENTICATION} -D ${DATADIR} --waldir=${POSTGRES_INITDB_WALDIR} ${INITDB_EXTRA_ARGS}"
+    command="$INITDB -U postgres --pwfile=<(echo "$POSTGRES_PASS") -E ${DEFAULT_ENCODING} --lc-collate=${DEFAULT_COLLATION} --lc-ctype=${DEFAULT_CTYPE} --wal-segsize=${WAL_SEGSIZE} --auth=${PASSWORD_AUTHENTICATION} -D ${DATADIR} ${INITDB_WALDIR_FLAG} ${INITDB_EXTRA_ARGS}"
+    echo "$command"
     su - postgres -c "$command"
+else
+    # If using existing datadir:
+    # Check if pg_wal symlink point to the correct directory described by POSTGRES_INITDB_WALDIR.
+    # Give warning if the value is not the same
+    if [[ -n "${POSTGRES_INITDB_WALDIR}" && \
+        "$(realpath ${POSTGRES_INITDB_WALDIR})" != "$(realpath \"$(readlink ${DATADIR}/pg_wal)\")" ]]; then
```

Comment:
`$(realpath \"$(readlink ${DATADIR}/pg_wal)\")` should be changed to

`$(realpath "$(readlink ${DATADIR}/pg_wal)")`

otherwise, you get the following error
![error](https://user-images.githubusercontent.com/2510900/130067877-1e950412-8791-4add-a2bb-d62cdff5e910.png)

Reply:
Ah, must have still pass the tests because it's just a warning and we don't check the else part.

Reply:
```suggestion
        "$(realpath ${POSTGRES_INITDB_WALDIR})" != "$(realpath "$(readlink ${DATADIR}/pg_wal)")" ]]; then
```